<launch>
    <node pkg="pointcloud_to_laserscan" type="pointcloud_to_laserscan_node" name="pointcloud_to_laserscan">
        <remap from="cloud_in" to="/points"/>
        <!-- <remap from="cloud_in" to="/unsw_vision/republisher/pcd"/> -->
        <remap from="scan" to="/scan_from_pcl"/>
        <rosparam>
            target_frame: "base_range_sensor_link"
            transform_tolerance: 0.01
            min_height: 0.0
            max_height: 1.10 # HSR height at normal pose

            angle_min:  -2.0987584590911865 # HSRB SPEC
            angle_max: 2.0987584590911865 # HSRB SPEC
            angle_increment: 0.004363323096185923 # HSRB SPEC
            scan_time: 0.02500000037252903 # HSR SPEC
            range_min:  0.019999999552965164 #HSR SEPC
            range_max: 60.0
            use_inf: true


            # Concurrency level, affects number of pointclouds queued for processing and number of threads used
            # 0 : Detect number of cores
            # 1 : Single threaded
            # 2->inf : Parallelism level
            concurrency_level: 1
        </rosparam>
    </node>
    
    <node name="laser_merger" pkg="nav_utils" type="laser_merger.py" output="screen">
    <param name = "laser1_topic" type = "str" value = "/hsrb/base_scan"/>
    <param name = "laser2_topic" type = "str" value = "/scan_from_pcl"/>
    </node>
    
</launch>